sphericart/CMakeLists.txt

```
@@ -13,7 +13,9 @@ option(BUILD_SHARED_LIBS "Build shared libraries instead of static ones" OFF)
 OPTION(SPHERICART_BUILD_TESTS "Build and run tests for Sphericart" OFF)
 OPTION(SPHERICART_OPENMP "Try to use OpenMP when compiling Sphericart" ON)
 OPTION(SPHERICART_ARCH_NATIVE "Try to use -march=native when compiling Sphericart" ON)
+option(SPHERICART_ENABLE_CUDA "Are we building the CUDA backend of Sphericart?" ON)
 
+set(CMAKE_CUDA_ARCHITECTURES native)
```

I'm a bit worried about setting this as the unconditional default, since it will produce binaries that only run on one system. Maybe we can re-use the same option as for CPU native arch?

```suggestion
if (SPHERICART_ARCH_NATIVE AND "${CMAKE_CUDA_ARCHITECTURES}" STREQUAL "")
    set(CMAKE_CUDA_ARCHITECTURES native)
endif()
```